src/NovelStepTimer.cpp

```diff
@@ -0,0 +1,109 @@
+//
+// Created by tagoo on 12/09/2019.
+//
+
+// This is based on the StepTimer provided in the DirectX ToolKit
+// Original code is available under the MIT License
+
+#include "NovelStepTimer.h"
+#include <SDL2/SDL.h>
+
+namespace NovelRT {
+  StepTimer::StepTimer(uint32_t targetFrameRate) :
+    _frequency(SDL_GetPerformanceFrequency()),
+    _maxCounterDelta(_frequency / 10), // 1/10th of a second
```

Comment:
It might be nice to make this configurable, it would need to be taken in as a `double maxSecondsDelta` and stored as ticks